frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/UserInfoSingleton.java

```diff
@@ -0,0 +1,207 @@
+        private HttpURLConnection connection;
+        private String token, serverResponse;
+        private String charset = java.nio.charset.StandardCharsets.UTF_8.name();
+        private boolean connectionError = false;
+        private String[] userInfo = new String[2];
+
+        @Override
+        protected Void doInBackground(Void... params) {
+            token = UserInfoSingleton.getInstance().getToken();
+            request();
+            return null;
+        }
+
+        private void request() {
+            String url = "http://128.199.167.57:8080/api/users/";
+
+            try {
+                connection = (HttpURLConnection) new URL(url).openConnection();
+                connection.setRequestMethod("GET");
+                connection.setRequestProperty("Authorization", "Bearer " + token);
+                connection.setRequestProperty("Accept-Charset", charset);
+
+                Log.i(TAG, "Token is " + token);
+
+                connection.connect();
+
+                Log.i(TAG, "Finished sending to server");
+
+                int status = connection.getResponseCode();
+                Log.i(TAG, "get user info status is " + status);
+
+                if (status == 200) {
+                    InputStream response = connection.getInputStream();
+                    // process the response
+                    BufferedReader br = new BufferedReader(new InputStreamReader(response));
+                    StringBuffer sb = new StringBuffer();
+                    Log.i(TAG, "Starting to read response");
+                    String line;
+                    while ((line = br.readLine()) != null) {
+                        sb.append(line + "\n");
+                        serverResponse = line;
+                        Log.i(TAG, "Registration Response>>>" + serverResponse);
+                    }
+                    try {
+                        JsonReader reader = new JsonReader(new StringReader(serverResponse));
+                        reader.beginObject();
+                        while (reader.hasNext()) {
+                            String readStr = reader.nextName();
+                            if (readStr.equals("email")) {
+                                userInfo[0] = reader.nextString();
+                                Log.i(TAG, userInfo[0]);
+                            } else if (readStr.equals("username")) {
+                                userInfo[1] = reader.nextString();
+                                Log.i(TAG, userInfo[1]);
+                            } else {
+                                reader.skipValue(); //avoid some unhandled events
+                            }
+                        }
+                        reader.endObject();
+                        reader.close();
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                } else if (status == 400) {
```

Comment:
400 and 401 are handled in the same way here, so should just write (status == 400 || status ==401)